backend/program/__init__.py

```diff
@@ -24,47 +46,209 @@ def __init__(self, args):
+            if service in get_args(Content):
+                if not getattr(item, 'title', None):
+                    # if we already have a copy of this item check if we've already recently 
+                    # updated the metadata
+                    if existing_item and not TraktMetadata.should_submit_item(existing_item):
+                        continue
+                    next_service = TraktMetadata
+                    self._submit_job(next_service, item)
+                    continue
+            
+            if service == TraktMetadata:
+                # grab a copy of the item in the container
+                if existing_item:
+                    # merge our fresh metadata item to make sure there aren't any
+                    # missing seasons or episodes in our library copy
+                    if isinstance(item, (Show, Season)):
+                        existing_item.fill_in_missing_info(item)
+                        existing_item.metadata_updated_at = item.metadata_updated_at
+                        item = existing_item
+                    # if after making sure we aren't missing any episodes check to 
+                    # see if we need to process this, if not then skip
+                    if existing_item.state == States.Library:
+                        logger.debug("%s is already complete and in the Library, skipping.", item.title)
+                        continue
+                next_service = Scraping
+                items_to_submit = [item]
+            elif service == Scraping:
+                # if we successfully scraped the item then send it to debrid
+                if item.state == States.Scrape:
+                    next_service = Debrid
+                    items_to_submit = [item]
+                # if we didn't get a stream then we have to dive into the components
+                # and try to get a stream for each one separately
+                elif isinstance(item, Show):
+                    next_service = Scraping
+                    items_to_submit = [s for s in item.seasons if s.state != States.Library]
+                elif isinstance(item, Season):
+                    next_service = Scraping
+                    items_to_submit = [e for e in item.episodes if e.state != States.Library]
+            elif service == Debrid:
+                next_service = Symlinker
+                if isinstance(item, Season):
+                    items_to_submit = [e for e in item.episodes]
+                elif isinstance(item, (Movie, Episode)):
+                    items_to_submit = [item]
+                items_to_submit = filter(lambda i: i.folder and i.file, items_to_submit)
+                if not items_to_submit:
+                    continue
+            elif service == Symlinker:
+                if item.symlinked:
+                    items_to_submit = [item]
+                next_service = PlexUpdater
+            elif service == PlexUpdater:
+                continue

```

Comment:
As we are still relying on plex I guess we should handle the plex updating part here?